strawberry_django/mutations/resolvers.py

```diff
@@ -99,8 +99,9 @@ def _parse_data(
                 continue
 
             if isinstance(v, ParsedObject):
-                if v.pk is None:
-                    v = create(info, model, v.data or {})  # noqa: PLW2901
+                if v.pk in (None, UNSET):  # noqa: PLR6201
+                    related_model = get_model_fields(model).get(k).related_model
+                    v = create(info, related_model, v.data or {})  # noqa: PLW2901
```

Comment:
Seems like `pk` may be either `UNSET` or `None`, in both cases it's necessary to create nested object. And to further properly perform `full_clean` and `save` we should identify related model (as in test type hierarchy most models use `name` clashing field names, cleaning could pass somehow, but finally object of different type could be created).